.circleci/config.yml

```
@@ -1,26 +1,95 @@
-# Use the latest 2.1 version of CircleCI pipeline process engine.
-# See: https://circleci.com/docs/2.0/configuration-reference
 version: 2.1
+orbs:
+  docker: circleci/docker@1.6.0
 
-# Define a job to be invoked later in a workflow.
-# See: https://circleci.com/docs/2.0/configuration-reference/#jobs
 jobs:
-  say-hello:
-    # Specify the execution environment. You can specify an image from Dockerhub or use one of our Convenience Images from CircleCI's Developer Hub.
-    # See: https://circleci.com/docs/2.0/configuration-reference/#docker-machine-macos-windows-executor
-    docker:
-      - image: cimg/base:stable
-    # Add steps to the job
-    # See: https://circleci.com/docs/2.0/configuration-reference/#steps
+  test_pytest:
+    docker: # executor type
+      - image: nipreps/miniconda:py39_2209.01
+        auth:
+          username: $DOCKER_USER
+          password: $DOCKER_PAT
+
+    environment:
+      - TEST_DIR: /tmp/tests
     steps:
       - checkout
       - run:
-          name: "Say hello"
-          command: "echo Hello, World!"
+          name: Get codecov
+          command: python -m pip install codecov
+
+      - run:
+          name: Run unit tests
+          no_output_timeout: 2h
+          command: |
+            export TEST_OUTPUT_DIR=${TEST_DIR}/output
+            mkdir -p ${TEST_OUTPUT_DIR} ${TEST_DIR}/summary ${TEST_DIR}/coverage
+            python -m pytest --junit-xml=${TEST_DIR}/summary/pytest.xml \
+                   --cov nireports --cov-report xml:${TEST_DIR}/coverage/unittests.xml \
+                   nireports/
+
+      - store_artifacts:
+          path: /tmp/tests/output
+
+      - store_test_results:
+          path: /tmp/tests/sumary/
```

Amend mistake in my last PR
```suggestion
          path: /tmp/tests/summary/
```